x-pack/plugins/endpoint/server/routes/resolver.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter } from 'kibana/server';
+import { SearchResponse } from 'elasticsearch';
+import { schema } from '@kbn/config-schema';
+import { EndpointAppContext, Total } from '../types';
+import {
+  getESChildrenQuery,
+  getESNodeQuery,
+  getESChildrenCountQuery,
+  getESNodeCountQuery,
+  PaginationInfo,
+} from '../services/resolver/query_builder';
+import { ResolverSearchHandler } from '../services/resolver/search_handler';
+import { ResolverData } from '../../common/types';
+import { EntityParseError } from '../services/resolver/common';
+
+export function registerResolverRoutes(router: IRouter, endpointAppContext: EndpointAppContext) {
+  const log = endpointAppContext.logFactory.get('resolver-route');
```

Comment:
Should we have each handler use its own log identifier, as used here, or should they all be `endpoint`? So far, I think I've just seen `endpoint`, so we'd probably want to change the other references if this is the way we want to go...

Reply:
I think having each handler use its own log identifier can give more context to the location of the message. They are reported in the logs like this:
`[18:35:11.867] [warning][endpoint][plugins][resolver-route] Error: Invalid entity_id received, must be`

Reply:
Can we just have it be `resolver` then? I feel the `route` part is unnecessary and redundant... in accordance, I'll change ours to `alerts` ...